### ISSUE TITLE
unable to split based on special charecters 

### ISSUE THREAD

**Issue by vemuriv** [ansible/ansible#36416]
verified
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

jinja2
##### ANSIBLE VERSION

```
ansible 2.5.0 (devel dc1b8e9a31) last updated 2017/12/21 00:40:43 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/ansible/ansible/lib/ansible
  executable location = /opt/ansible/ansible/bin/ansible
  python version = 2.7.13 (default, Jan 11 2017, 10:56:06) [GCC]

```

##### CONFIGURATION

HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False

##### OS / ENVIRONMENT

SLES12
##### SUMMARY

unable to split based on '\\'

##### STEPS TO REPRODUCE

unable to split based on '\'

```yaml
vars:
      source_instance: "abcd\\efgh"
          s_instance: ""

- set_fact:
         s_instance: '{{ source_instance.split( "\\")[1] }}'

```

##### EXPECTED RESULTS

efgh

  ##### ACTUAL RESULTS

TASK [set_fact] *********************************************************************************************************************************************************
task path: /root/ansible/examples/wintest4.yml:23
fatal: [xxxxxx]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: list object has no element 1\n\nThe error appears to have been in '/root/ansible/examples/wintest4.yml': line 23, column 8, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n     - set_fact:\n       ^ here\n"
}

```
 "msg": "The task includes an option with an undefined variable. The error was: list object has no element 1\n\nThe error appears to have been in '/root/ansible/examples/wintest4.yml': line 23, column 8, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n     - set_fact:\n       ^ here\n"
task path: /root/ansible/examples/wintest4.yml:13
ok: [cdlddshd01] => {
    "msg": " value for source is  abcd\\efgh"
}

TASK [set_fact] *********************************************************************************************************************************************************
task path: /root/ansible/examples/wintest4.yml:23
fatal: [cdlddshd01]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: list object has no element 1\n\nThe error appears to have been in '/root/ansible/examples/wintest4.yml': line 23, column 8, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n     - set_fact:\n       ^ here\n"
}

```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/template](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by vemuriv** [comment]
Is this waiting on some input from me ?   Please let me know 
Thanks

**Comment by Akasurde** [comment]
@vemuriv You may want to store special character in vars and then use it like -

```yaml
---
- hosts: localhost
  vars:
    source_instance: "abcd\\efgh"
    separator: "\\"
  tasks:
    - debug:
        msg: '{{ source_instance.split(separator) }}'
```

**Comment by Akasurde** [comment]
needs_info

**Comment by ansibot** [comment]
@vemuriv This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@vemuriv You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
